x-pack/plugin/src/test/resources/rest-api-spec/test/ml/get_datafeed_stats.yml

```diff
@@ -158,6 +164,40 @@ setup:
   - is_true: datafeeds.0.node.transport_address
   - match: { datafeeds.0.node.attributes.ml\.max_open_jobs: "20"}
 
+---
+"Test get stats for started datafeed contains timing stats":
+
+  - do:
+      ml.open_job:
+        job_id: get-datafeed-stats-1
+
+  - do:
+      ml.start_datafeed:
+        datafeed_id: "datafeed-1"
+        start: 0
+
+  - do:
+      ml.get_datafeed_stats:
+        datafeed_id: datafeed-1
+  - match: { datafeeds.0.datafeed_id: "datafeed-1"}
+  - match: { datafeeds.0.state: "started"}
+  - match: { datafeeds.0.timing_stats.job_id: "get-datafeed-stats-1"}
+  - match: { datafeeds.0.timing_stats.search_count: 0}
+  - match: { datafeeds.0.timing_stats.total_search_time_ms: 0.0}
+
+  - do:
+      ml.stop_datafeed:
+        datafeed_id: "datafeed-1"
+
+  - do:
+      ml.get_datafeed_stats:
+        datafeed_id: datafeed-1
+  - match: { datafeeds.0.datafeed_id: "datafeed-1"}
+  - match: { datafeeds.0.state: "stopped"}
+  - match: { datafeeds.0.timing_stats.job_id: "get-datafeed-stats-1"}
+  - match: { datafeeds.0.timing_stats.search_count: 1}
```

Comment:
Is there something that ensures that the timing stats document is searchable in the index after stopping a datafeed?  I couldn't see this but maybe I'm just missing it (in which case the rest of this comment is irrelevant).

In other words, what stops the following sequence of events?

1. Datafeed is started
2. Datafeed does a search
3. Datafeed indexes a timing stats document, which sits in the translog and is not yet committed to Lucene
4. Datafeed is stopped
5. Get datafeed stats searches the index, and does not find the most recent timing stats document containing details of the search
6. Translog is flushed and the latest timing stats becomes searchable

The test might be asserting in between steps 5 and 6.  Of course, it's _possible_ that the document becomes searchable before the get datafeed stats request runs, so then the test will pass.

There are two approaches for solving this:

1. On stopping a datafeed persist the most recent timing stats again with refresh=immediate.
2. Use refresh=immediate every time datafeed timing stats are persisted.

I prefer option 1 but if it's really hard then option 2 is probably OK as the volume is not that high.

Reply:
I didn't want to make this (already big) PR any bigger so I implemented approach 2. and added TODO to revisit in the future.
Is that ok?

BTW, I was not aware of the translog